src/extrapolation/filled.jl

```diff
@@ -60,7 +60,7 @@ end
         gradient(itp, x...)
     else
         e = eltype(itp)
-        ptype = promote_type(e, typeof(etp.fillvalue))
+        ptype = promote_type(e, mapreduce(typeof, promote_type, x))
```

Comment:
Can't we let Julia's type system figure out the types? Arebts things like `itp` already parameterized?

Reply:
Not really, this is because interpolations gradient use `x`(the indices) to generate `δx` and thus the gradient output, to keep type stability we need to keep consistent to interpolation gradient internal.